src/components/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,51 @@
+/// <reference types="vite-plugin-svgr/client" />
+import Check from '../../assets/icons/check.svg?react';
+import { Minus } from '@phosphor-icons/react';
+
+interface CheckboxComponentProps {
+  checked?: boolean;
+  indeterminate?: boolean;
+  disabled?: boolean;
+  onClick?: () => void;
+  required?: boolean;
+}
+
+const Checkbox = ({
+  checked = true,
+  indeterminate = false,
+  disabled = false,
+  onClick,
+  required,
+}: CheckboxComponentProps): JSX.Element => {
+  return (
+    <>
+      <button
+        onClick={(e) => {
```

Comment:
Maybe you can pass onClick here directly and define it in the props as `onClick: (e: unknown) => void;`. This way, we modify the onClick outside the component and not add logic inside the components.

Reply:
Done